apps/sms/test/marionette/lib/messages.js

```diff
@@ -256,6 +256,34 @@
           client.helper.waitForElement(SELECTORS.main);
         },
 
+        launchFromNotification: function(notification) {
+          var chromeClient = client.scope({ context: 'chrome' });
+          chromeClient.executeScript(function(origin, notification) {
+            /* global Components, Services */
+            var managerClass = Components.classes[
+              '@mozilla.org/system-message-internal;1'
+            ];
+
+            var pageURI = Services.io.newURI(
+              origin + '/index.html#notification', null, null
+            );
+
+            var manifestURI = Services.io.newURI(
+              origin + '/manifest.webapp', null, null
+            );
+
+            var systemMessageManager = managerClass.getService(
+              Components.interfaces.nsISystemMessagesInternal
+            );
+
+            systemMessageManager.sendMessage(
+              'notification', notification, pageURI, manifestURI
+            );
+          }, [ORIGIN_URL, notification]);
+
+          this.switchTo();
+        },
+
```

Comment:
+ ideally marionette should provide smth like ```launchFromSystemMessage``` since it's declared in app manifest